tests/test_serial_cc.py

```diff
@@ -37,3 +38,35 @@ def test_serial_cryptocontext(tmp_path):
     assert isinstance(ct1, fhe.Ciphertext)
     LOGGER.debug("Cryptocontext deserializes to %s %s", success, ct1)
     assert fhe.SerializeToFile(str(tmp_path / "ciphertext12.json"), ct1, fhe.JSON)
+
+
+@pytest.mark.parametrize("mode", [fhe.JSON, fhe.BINARY])
```

Comment:
Please also add (de)serialization for eval and public keys so all methods could be tested in the unit test. Use simple-integers-serial.py as the template.

Reply:
Tests added.

Since the serialization/deserialization of EvalKey was made for PRE keys, I've added it like that. I think EvalMult and EvalAutomorphism would need their own functions to work.

Reply:
I would like to confirm. Is your current assumption that EvalMult and EvalAutomorphism keys would be serialized and deserialized using files, while other crypto objects could be serialized as Bytes/Strings. These keys are needed to perform any homomorphic multiplication or rotation operation. Just trying to confirm what the goal of the PR is. Is the goal of the PR to only support PKE, PRE, and additive HE w/o supporting SHE or FHE yet?

Reply:
Yes, that's correct. I was trying to address the most common use cases that could arise with serialization and deserialization for my organization and didn't take SHE and FHE into account.

I could add specific functions for those use cases, if necessary

Reply:
Thanks. These can be added in a separate PR.